[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/miguepoloc/toma-decisiones-mcda/blob/main/03_vikor_cacao.ipynb)

# VIKOR, caso cacao

Mismo caso y mismos datos que TOPSIS. Contenido completo en la Sesión 3 (TOPSIS/VIKOR) del curso.

In [1]:
!pip install -q pyDecision


[notice] A new release of pip is available: 26.1.2 -> 26.2.1
[notice] To update, run: pip install --upgrade pip


In [2]:
import numpy as np
from pyDecision.algorithm import vikor_method

zonas = ["Bonda", "Guachaca", "San Pedro", "Palmor"]
dataset = np.array([
    [27, 78, 6.2, 0.9],
    [26, 85, 5.8, 1.5],
    [24, 82, 6.5, 0.7],
    [22, 88, 5.5, 1.2],
])
pesos = [0.25, 0.30, 0.25, 0.20]
tipo = ['max', 'min', 'max', 'min']

## Cálculo de S, R, Q

In [3]:
s, r, q, solution = vikor_method(dataset, pesos, tipo,
                                  strategy_coefficient=0.5,
                                  graph=False, verbose=False)
print("S:\n", s)
print("R:\n", r)
print("Q:\n", q)

S:
 [[1.    0.125]
 [3.    0.27 ]
 [2.    0.635]
 [4.    0.925]]
R:
 [[1.    0.075]
 [3.    0.15 ]
 [2.    0.21 ]
 [4.    0.3  ]]
Q:
 [[1.         0.        ]
 [3.         0.25729167]
 [2.         0.61875   ]
 [4.         1.        ]]


**Resultado esperado** (coincide con lo publicado en la Sesión 3 del curso): Bonda S=0.125/R=0.075/Q=0.000, San Pedro S=0.270/R=0.150/Q=0.257, Guachaca S=0.635/R=0.210/Q=0.619, Palmor S=0.925/R=0.300/Q=1.000. Menor Q es mejor → 1º Bonda, 2º San Pedro, 3º Guachaca, 4º Palmor.